Remove code injection. Use message passing instead.

rchrd2 opened the pull request.

wagnerand commented:
You want to make sure the url is actually a valid url, and not some `javascript:...` which would allow remote code injection with XSS again.

wagnerand commented on src/scripts/client.js:
```diff
@@ -171,26 +175,28 @@
+  function checkIt(wayback_url) {
+    // Some pages use javascript to update the dom so poll to ensure
+    // the banner gets recreated if it is deleted.
+    enforceBannerInterval = setInterval(function() {
```
I know that's not new code, but you want to cancel the timer in `onunload`.

rchrd2 replied:
I can add that, but what's the benefit?

wagnerand replied:
It's generally considered good practice to clean up after yourself. It also prevents memory leaks, although I think if the page object in memory dies, the timer would die as well.

rchrd2 commented:
@wagnerand I updated it to have URL validation like you suggested. I moved that code into it's own function. Let me know if you notice anything else. Thanks.

wagnerand commented on src/scripts/background.js:
```diff
@@ -44,21 +46,44 @@ function wmAvailabilityCheck(url, onsuccess, onfail) {
+ */
+function isValidSnapshotUrl(url) {
+  return ((typeof url) === "string" &&
+    (url.indexOf("http://") === 0 || url.indexOf("https://") === 0));
```
Nitpick: This could be `/^https?:\/\//.test(url)`.

wagnerand commented:
Looks great, thank you!
